UTSAVS26/PyVerse
# Added Insertion Sort Visualizer Project in Web _Development folder

## Pull Request for PyVerse 💡

### Requesting to submit a pull request to the PyVerse repository.

---

#### Issue Title
Added Insertion Sort Visualizer Project in Web _Development folder

---

**What's the goal of the project?**  
The aim of the project is to create a visual representation of the insertion sort algorithm. This web-based application helps users understand the working of insertion sort through step-by-step visualization. It allows users to:

- Add numbers dynamically to form an array.
- Visualize the sorting process as the algorithm sorts the array, showing how elements are compared, moved, and inserted in their correct positions.
- Observe each step, such as selecting a key element, comparing it with previous elements, and shifting elements to make space for the key.
- Reset the data to allow multiple attempts or learning sessions.


---

#### Name
Ananya Gupta

---

#### GitHub ID
https://github.com/ananya8606

---

#### Email ID
www.aynanagupta@gmail.com

---

#### Identify Yourself
 GSSOC contributer 


---

#### Type of Change
- New feature (non-breaking change which adds functionality)

---

#### How Has This Been Tested?
The project has been tested through the following methods:

Manual Testing:

1. Basic Functionality: Verified that users can add numbers to the array, trigger the sorting process, and reset the data. Ensured that the sorting animation progresses correctly.
2. Edge Cases: Tested scenarios such as:

-  Sorting an empty array.
-  Sorting an array with a single element.
- Sorting arrays with duplicate values.

3. User Interaction: Checked the responsiveness of user controls, including buttons for adding numbers, sorting, and resetting.
4. Algorithm Accuracy:

- Compared the output of the visualized insertion sort with the expected sorted array to ensure the algorithm correctly sorts the array in all cases.
- Verified that the sorting steps follow the correct sequence of comparisons and movements as defined by the insertion sort algorithm.

Cross-Browser Testing:

Tested the web application on different browsers (e.g., Chrome, Firefox, Edge) to ensure consistent behavior and compatibility.

UI/UX Testing:

1. Assessed the clarity of the visualization, ensuring that steps in the sorting process are easy to follow.
2. Verified that the application remains responsive and does not freeze or behave unexpectedly during sorting.

---

#### Checklist
**Please confirm the following:**  
-  My code follows the guidelines of this project.
- I have performed a self-review of my own code.
-  I have commented my code, particularly wherever it was hard to understand.
- I have made corresponding changes to the documentation.
-  My changes generate no new warnings.
-  I have added things that prove my fix is effective or that my feature works.
- Any dependent changes have been merged and published in downstream modules.
